### PR TITLE
Adding TCN decoder.

### PR DESCRIPTION
**Pull request by hsezhiyan**

### CODE REVIEW

File: oatomobile/baselines/torch/dim/train.py [hunk 1]
```diff
@@ -15,6 +15,7 @@
 """Trains the deep imitative model on expert demostrations."""
 
 import os
+import pdb
```
Comment (filangelos) [inline oatomobile/baselines/torch/dim/train.py]: remove `pdb` from the list of imports

File: oatomobile/baselines/torch/dim/model.py [hunk 2]
```diff
@@ -31,6 +31,7 @@
 from oatomobile.torch.networks.mlp import MLP
 from oatomobile.torch.networks.perception import MobileNetV2
 from oatomobile.torch.networks.sequence import AutoregressiveFlow
+from oatomobile.torch.networks.tcn import TCN
```
Comment (filangelos) [inline oatomobile/baselines/torch/dim/model.py]: add `TCN` in `oatomobile.torch.networks.sequence`

File: oatomobile/baselines/torch/dim/train.py [hunk 3]
```diff
@@ -180,11 +187,13 @@ def train_step(
   ) -> torch.Tensor:
     """Performs a single gradient-descent optimisation step."""
     # Resets optimizer's gradients.
+    # pdb.set_trace()
```
Comment (filangelos) [inline oatomobile/baselines/torch/dim/train.py]: remove any `pdb` references

File: oatomobile/baselines/torch/dim/train.py [hunk 4]
```diff
@@ -180,11 +187,13 @@ def train_step(
...
     optimizer.zero_grad()
 
+    target_mean = batch["player_future"][..., :2] 
     # Perturb target.
     y = torch.normal(  # pylint: disable=no-member
-        mean=batch["player_future"][..., :2],
+        mean=target_mean,
         std=torch.ones_like(batch["player_future"][..., :2]) * noise_level,  # pylint: disable=no-member
```
Comment (filangelos) [inline oatomobile/baselines/torch/dim/train.py]: use `target_mean` instead of `batch["player_future"][..., :2]`

File: oatomobile/baselines/torch/dim/train.py [hunk 5]
```diff
@@ -195,10 +204,15 @@ def train_step(
         is_at_traffic_light=batch["is_at_traffic_light"],
         traffic_light_state=batch["traffic_light_state"],
     )
-    _, log_prob, logabsdet = model._decoder._inverse(y=y, z=z)
 
-    # Calculates loss (NLL).
-    loss = -torch.mean(log_prob - logabsdet, dim=0)  # pylint: disable=no-member
+    if use_tcn:
```
Comment (filangelos) [inline oatomobile/baselines/torch/dim/train.py]: the API for the `decoder` should be the same, regardless if it's an RNN or a TCN

File: oatomobile/baselines/torch/dim/train.py (hunk 4 above)
Comment (hsezhiyan) [reply oatomobile/baselines/torch/dim/train.py]: @filangel it looks like I'm already using `target_mean`. Can you clarify?

File: oatomobile/baselines/torch/dim/train.py (hunk 3 above)
Comment (hsezhiyan) [reply oatomobile/baselines/torch/dim/train.py]: Resolved.

File: oatomobile/baselines/torch/dim/model.py (hunk 2 above)
Comment (hsezhiyan) [reply oatomobile/baselines/torch/dim/model.py]: Resolved.

File: oatomobile/baselines/torch/dim/train.py (hunk 1 above)
Comment (hsezhiyan) [reply oatomobile/baselines/torch/dim/train.py]: Resolved.

File: oatomobile/baselines/torch/dim/train.py (hunk 5 above)
Comment (hsezhiyan) [reply oatomobile/baselines/torch/dim/train.py]: Can you clarify what you mean by this? By API, do you want the TCN decoder to similarly have a `.inverse` method, as the `Autoregressive` decoder does?
